test/functional/tests/performance/test_performance_writes.py

```diff
@@ -0,0 +1,162 @@
+# SPDX-License-Identifier: BSD-3-Clause
+#
+from datetime import timedelta
+
+import pytest
+
+from api.cas import casadm
+from api.cas.cache_config import CacheLineSize, CacheMode, SeqCutOffPolicy
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import ReadWrite, IoEngine, CpusAllowedPolicy
+from test_utils.os_utils import (
+    set_wbt_lat,
+    get_number_of_processors_from_cpuinfo,
+    get_dut_cpu_physical_cores,
+)
+from test_utils.size import Unit, Size
+
+
+# TODO: for disks other than Intel Optane, fio ramp is needed before fio tests on raw disk
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrizex("block_size", [Size(1, Unit.Blocks4096), Size(8, Unit.Blocks4096)])
+@pytest.mark.parametrizex("queue_depth", [1, 16, 32])
+def test_performance_write_hit_wb(block_size, queue_depth):
+    """
+    title: Test Open CAS performance for 100% write inserts scenario in write-back mode.
+    description: |
+        Compare write hit performance (throughput and latency) for Open CAS vs raw device
+        for different start command options. Open CAS in Write-Back mode device should
+        provide comparable throughput to bare cache device.
+    pass_criteria:
+      - passes performance threshold
+    """
+
+    processors_number = get_number_of_processors_from_cpuinfo()
+    num_jobs = [int(processors_number / 2), processors_number]
+    data_size = Size(20, Unit.GibiByte)
+    cache_size = Size(24, Unit.GibiByte)
+    cache_line_size = CacheLineSize.LINE_4KiB
+    raw_disk_results = {}
+    cas_results = {}
+
+    fio_command = (
+        Fio()
+        .create_command()
+        .direct()
+        .read_write(ReadWrite.randrw)
+        .write_percentage(100)
+        .io_engine(IoEngine.libaio)
+        .cpus_allowed(get_dut_cpu_physical_cores())
+        .cpus_allowed_policy(CpusAllowedPolicy.split)
+        .run_time(timedelta(seconds=450))
+        .block_size(block_size)
+        .io_depth(queue_depth)
+    )
+
+    with TestRun.step("Prepare partitions for cache and core"):
+        cache_device = TestRun.disks["cache"]
+        cache_device.create_partitions([cache_size])
+        cache_part = cache_device.partitions[0]
+
+        core_device = TestRun.disks["core"]
+        core_device.create_partitions([data_size])
+        core_part = core_device.partitions[0]
+
+    with TestRun.step("Measure 4KiB write performance (throughput and latency) on raw disk."):
+        fio_command.target(cache_part)
+        for nj in num_jobs:
+            # jobs directed to specific offsets
+            offset = data_size.value // nj
+            # round down offset to multiplication of 512 blocks
+            offset = Size(offset // Unit.Blocks512.get_value() * Unit.Blocks512.get_value())
```

Comment:
I'm not sure, but do we still support 512B? Should it be 4KiB instead?

Reply:
both are still supported :)